Tests/SwiftProtobufTests/Test_FuzzTests.swift

```diff
@@ -0,0 +1,124 @@
+// decodes doesn't matter, just that it can/can't decode without
+//
+// If a failure is found via manually running the binaries, the failures end
+// with a summmary that shows the data output is hex pairs, an escaped string,
+// written to a file, and then also as Base64. You can directly get the hex
+// values from there, or can use xxd to dump from the file.
+//
+// For failures that get filed by oss-fuzz, the bugs will have links to detailed
+// reports, those have a test case (and reduced test case) that can be download.
+//
+// Easily get the hex dumps via `xxd -i < _test_case_file_`
+
+import Foundation
+import XCTest
+
+class Test_FuzzTests: XCTestCase {
+
+  func assertBinaryFails(_ bytes: [UInt8], file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
+      let _ = try Fuzz_Testing_Message(serializedData: Data(bytes), extensions: Fuzz_Testing_FuzzTesting_Extensions)
+      XCTFail(file: file, line: line)
+    } catch {
+      // Error while parsing is expected.
+    }
+  }
+
+  func assertJSONFails(_ jsonBytes: [UInt8], file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
+      let _ = try Fuzz_Testing_Message(jsonUTF8Data: Data(jsonBytes), extensions: Fuzz_Testing_FuzzTesting_Extensions)
+      XCTFail(file: file, line: line)
+    } catch {
+      // Error while parsing is expected.
+    }
+  }
+
+  func assertJSONFails(_ json: String, file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
+      let _ = try Fuzz_Testing_Message(jsonString: json, extensions: Fuzz_Testing_FuzzTesting_Extensions)
+      XCTFail(file: file, line: line)
+    } catch {
+      // Error while parsing is expected.
+    }
+  }
+
+  func assertTextFormatFails(_ textFormat: String, file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
+      let _ = try Fuzz_Testing_Message(textFormatString: textFormat, extensions: Fuzz_Testing_FuzzTesting_Extensions)
+      XCTFail(file: file, line: line)
+    } catch {
+      // Error while parsing is expected.
+    }
+  }
+
+  func assertTextFormatFails(_ asBytes: [UInt8], file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
+      guard let str = String(data: Data(asBytes), encoding: .utf8) else {
+        XCTFail(file: file, line: line)
+        return
+      }
+      let _ = try Fuzz_Testing_Message(textFormatString: str, extensions: Fuzz_Testing_FuzzTesting_Extensions)
+      XCTFail(file: file, line: line)
+    } catch {
+      // Error while parsing is expected.
+    }
+  }
+
+  func assertTextFormatSucceeds(_ textFormat: String, file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
+      let _ = try Fuzz_Testing_Message(textFormatString: textFormat, extensions: Fuzz_Testing_FuzzTesting_Extensions)
+    } catch {
+      XCTFail(file: file, line: line)
+    }
+  }
+
+  func assertTextFormatSucceeds(_ asBytes: [UInt8], file: XCTestFileArgType = #file, line: UInt = #line) {
+    do {
```

Comment:
And this could be

```swift
let str = XCTUnwrap(String(...), file: file, line: line)
XCTAssertNoThrow(try Fuzz_Testing_Message(...), file: file, line: line)
```

(But `XCTUnwrap` is only Xcode 11.0+, in case we're still supporting anything older.)

Reply:
we still support/test with Swift 4.2, so…